File: go/random-int-between.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	fmt.Println(randomInt(1, 10))
}

// generate a random integer
func randomInt(min, max int) int {
	rand.Seed(time.Now().UTC().UnixNano())
	return rand.Intn(max-min) + min
}

File: TypeScript/Random/random-int-between.ts
import * as crypto from "crypto"

function main() {
    // Generate a random number between 1 and 100
    console.log(randomIntBetween(1, 100))
}

main()

// Generate a random integer between min (inclusive) and max (exclusive)
function randomIntBetween(min: number, max: number): number {
    return crypto.randomInt(max - min) + min
}